### PR TITLE
Implement ADC for Nuttx

### PR DESCRIPTION
**Pull request by hs0225**
Related issue: #578 
- ADC API for Nuttx
   - constructor(export)
   - read
   - readSync
   - unexport

- Add ADC pin
- Modify Pin module : fix typing error, rename pin object

IoT.js-DCO-1.0-Signed-off-by: Hosung Kim hs852.kim@samsung.com

### CODE REVIEW

File: src/module/iotjs_module_pin.h [hunk 1]
```diff
@@ -17,13 +17,30 @@
...
-#define TIMER_PIN_MASK 15
-#define TIMER_NUM(n) ((n) << TIMER_PIN_SHIFT)
 
-#endif
+#if ENABLE_MODULE_ADC || ENABLE_MODULE_PWM
+
+#define SYSIO_TIMER_PIN_SHIFT 21 /* Bits 21-24: Timer number */
+#define SYSIO_TIMER_PIN_MASK 15
+#define SYSIO_TIMER_NUMBER(n) ((n) << SYSIO_TIMER_PIN_SHIFT)
+#define SYSIO_GET_TIMER(n) ((n >> SYSIO_TIMER_PIN_SHIFT) & SYSIO_TIMER_PIN_MASK)
```
Comment (glistening) [inline src/module/iotjs_module_pin.h]: Please put parentheses () around n.

File: src/module/iotjs_module_pin.h [hunk 2]
```diff
@@ -17,13 +17,30 @@
...
+
+#endif /* ENABLE_MODULE_ADC || ENABLE_MODULE_PWM */
+
+
+#if ENABLE_MODULE_ADC
+
+#define ADC_NUMBER_SHIFT 25 /* Bits 25-26: ADC number */
+#define ADC_NUMBER_MASK 3
+#define ADC_NUMBER(n) ((n) << ADC_NUMBER_SHIFT)
+#define ADC_GET_NUMBER(n) ((n >> ADC_NUMBER_SHIFT) & ADC_NUMBER_MASK)
```
Comment (glistening) [inline src/module/iotjs_module_pin.h]: ditto

File: src/module/iotjs_module_pin.h (hunk 1 above)
Comment (hs0225) [inline src/module/iotjs_module_pin.h]: I fixed it. Thanks for your review.

**Comment by nova0821** [comment]
Please change ``function (ARGS)`` to ``function(ARGS)`` in a javascript file.

**Comment by hs0225** [comment]
@nova0821 I updated as your comment.

**Comment by nova0821** [comment]
LGTM

**Pull request merged**